components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/graphQL/analytics/GraphQLOperationInfoAnalyzer.java

```diff
@@ -0,0 +1,89 @@
+/*
+ *  Copyright (c) 2024, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.apimgt.gateway.handlers.graphQL.analytics;
+
+
+import graphql.language.AstPrinter;
+import graphql.language.Definition;
+import graphql.language.Document;
+import graphql.language.Field;
+import graphql.language.OperationDefinition;
+import graphql.language.Selection;
+import graphql.parser.Parser;
+import graphql.schema.GraphQLSchema;
+import graphql.schema.idl.TypeDefinitionRegistry;
+import org.apache.synapse.MessageContext;
+import org.wso2.carbon.apimgt.common.gateway.graphql.GraphQLOperationAnalyzer;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+
+/**
+ * analyze each query and mutation
+ * then engages metric handler to handle metric publishing
+ */
+public class GraphQLOperationInfoAnalyzer {
+
+    private GraphQLAnalyticsMetricsHandler graphQLMetricsHandler;
+
+    public void analyzePayload(MessageContext messageContext){
+
+        Parser parser = new Parser();
+        String payload = (String) messageContext.getProperty(APIConstants.GRAPHQL_PAYLOAD);
```

Comment:
We do not need repetitive work here as we have already extract the payload at APIGraphQLHandler. You can do payload scanning at there and set them into synapse properties.

Reply:
Fixed via https://github.com/wso2/carbon-apimgt/pull/12441/commits/226c49a261240dc5305f96d69aadb6f86edfae84